Objects/obmalloc.c

```diff
@@ -1233,6 +1243,259 @@ _Py_GetAllocatedBlocks(void)
     return n;
 }
 
+/*==========================================================================*/
+/* radix tree for tracking arena usage
+
+   bit allocation for keys (2^20 arena size)
+
+   64-bit pointers:
+     16 -> ignored (BITS - PHYSICAL_BITS)
+     10 -> MAP_TOP
+     10 -> MAP_MID
+      8 -> MAP_BOT
+     20 -> ideal aligned arena
+   ----
+     64
+
+   32-bit pointers:
+     12 -> MAP_BOT
+     20 -> ideal aligned arena
```

Comment:
18?

Reply:
I believe the numbers are correct for 2^20 size pools.  However, I changed to code to use 2^18 sized pools if SIZEOF_VOID_P <= 4, on 32-bit platform.  So, maybe I should change the example to match, e.g.
```
   64-bit pointers, for 2^20 arena size:
     16 -> ignored (BITS - PHYSICAL_BITS)
     10 -> MAP_TOP
     10 -> MAP_MID
      8 -> MAP_BOT
     20 -> ideal aligned arena
   ----
     64

   32-bit pointers, for 2^18 arena size:
     14 -> MAP_BOT
     18 -> ideal aligned arena
   ----
     32
```